### ISSUE TITLE
Staging Review finding: "Filter and sort" link, multiple accessibility barriers

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#98565]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Transition Experience
**Product:** Transition Experience MVP v1
**Feature:** V1 Transition Experience

## Findings details
**VA.gov Experience Standard - issue:** [User experiences an unexpected interaction with a component or pattern.](https://design.va.gov/about/experience-standards/#user-experiences-predictable-interactions-with-components)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

On the results page at high browser zoom or on mobile devices, there's a "Filter and sort" link (rather than showing the filter/sort options by default). Clicking "Filter and sort" reveals an expandable section of content with the filter/sort options.

There are a number of issues with the way this is implemented:
- Since the "Filter and sort" link doesn't have an `href`, browsers don't make it keyboard focusable by default. For a low vision user at high browser zoom who primarily uses a keyboard, there's no way to get to the link.
- Clicking the "Filter and sort" link displays and hides the content, but there's no notification provided to screen reader users that a change has occurred --- just silence. Typically, when a screen reader user interacts with a toggle like this the expectation is that it announces the current state of the content it controls (expanded or collapsed).
- The "Filter and sort" link doesn't navigate anywhere --- so it shouldn't be a link, and shouldn't be a link with `role="button"` applied to it either. Links and buttons have different interaction patterns for some types of assistive technology, and a link that does button-y things or a button that does link-y things can be confusing (for example, a sighted voice command software user may visually identify it as a link but be unable to interact with it via common link interactions). Wherever possible we try to maintain material honesty, meaning: a link goes somewhere, looks like a link, and is coded as a link; a button does an action, looks like a button, and is coded as a button.

<details><summary>Link, screenshot or steps to recreate</summary>

The element is currently coded as:
```
<va-link-action text="Filter and sort" type="secondary" role="button" class="hydrated"></va-link-action>
```

... which renders as:
```
<a class="va-link--secondary">
   <va-icon class="link-icon--left link-icon hydrated"></va-icon>
   <span class="link-text">Filter and sort</span>
</a>
```

</details>

## Recommended action

The easiest option here is to use the `va-accordion` component, which has all of the required accessibility support for expanding/collapsing a region of content. If a button is preferable, then I'd recommend looking at the `va-accordion` component and matching its accessibility features (eg. using `aria-expanded` appropriately). But it definitely shouldn't be a link it if it's not going somewhere new.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Keyboard
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by nkontrabecki** [comment]
Good Morning @briandeconinck Thank you for the feedback! Is it acceptable for us to remove the Filter and Sort link and make the Filter and Sort always visible in high browser zoom or on mobile device?

**Comment by briandeconinck** [comment]
Hi @nkontrabecki! Yes --- always visible at high zoom / mobile device is totally fine here.

**Comment by nkontrabecki** [comment]
Addressed in [Jira sprint ticket 359](https://jira.devops.va.gov/browse/PTEMSVT-359)in Sprint 17. @briandeconinck, please let us know if this meets the requirement. [Our Staging environment](https://staging.va.gov/discover-your-benefits)

**Comment by nkontrabecki** [comment]
Confirmed fixed and tested in sprint 17 Jira ticket 359. Brian Deconinck reviewed and confirmed. Thank you!
